sklearn/cluster/_kmeans.py

```diff
@@ -1188,9 +1188,8 @@ class KMeans(_BaseKMeans):
             (n_clusters, n_features), default='k-means++'
         Method for initialization:
 
-        'k-means++' : selects initial cluster centers for k-mean
-        clustering in a smart way to speed up convergence. See section
-        Notes in k_init for more details.
+        'k-means++' : selects initial cluster centers for k-mean clustering in a
+        smart way to speed up convergence. See notes in `n_init` for more details.
```

Comment:
The previous "in in a smart way" is not really descriptive. Should we put more details in the description of `k-means++`?
```suggestion
        'k-means++' : selects initial cluster centroids using sampling 
        based on a empirical probably distribution of the points
        contribution to the overall inertia.
        This technique speeds up convergence, and is theoretically
        proven to be :math:`\mathcal{O}(\log k)`-optimal.
        See the description of `n_init` for more details.
```

Also, would it be useful to add [k-means++ original paper](https://dl.acm.org/doi/10.5555/1283383.1283494) to the references?

---

The same suggestion apply in the other docstrings.